src/CR.WorkingDayService/WorkingDayServiceBuilder.cs

```diff
@@ -0,0 +1,96 @@
+﻿// <copyright file="WorkingDayServiceBuilder.cs" company="Cognisant">
+// Copyright (c) Cognisant. All rights reserved.
+// </copyright>
+
+namespace CR.WorkingDayService
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// A Builder Class for the Working Day Service; used for easy configuration.
+    /// </summary>
+    public sealed class WorkingDayServiceBuilder
+    {
+        private readonly HashSet<IWorkingDaySource> _sources;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WorkingDayServiceBuilder"/> class.
+        /// </summary>
+        public WorkingDayServiceBuilder() => _sources = new HashSet<IWorkingDaySource>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WorkingDayServiceBuilder"/> class.
+        /// </summary>
+        /// <returns>The new instance of the <see cref="WorkingDayServiceBuilder"/> class.</returns>
+        public static WorkingDayServiceBuilder New() => new WorkingDayServiceBuilder();
+
+        /// <summary>
+        /// Builds a <see cref="WorkingDayService"/> with the configured set of <see cref="IWorkingDaySource"/>s.
+        /// </summary>
+        /// <returns>A <see cref="WorkingDayService"/> which uses the <see cref="IWorkingDaySource"/>s in the <see cref="WorkingDayServiceBuilder"/>'s <c>_sources</c>.</returns>
+        public WorkingDayService Build() => new WorkingDayService(_sources.ToList());
+
+        /// <summary>
+        /// Configure the <see cref="WorkingDayServiceBuilder"/> to use the specified <see cref="IWorkingDaySource"/> in addition to its current sources.
+        /// </summary>
+        /// <param name="source">The <see cref="IWorkingDaySource"/> to add.</param>
+        /// <returns>The same instance of the <see cref="WorkingDayServiceBuilder"/> with all of its current sources, and the new <see cref="IWorkingDaySource"/> which was passed in.</returns>
+        public WorkingDayServiceBuilder AddSource(IWorkingDaySource source)
+        {
+            _sources.Add(source);
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the <see cref="WorkingDayServiceBuilder"/> to only use the passed in <see cref="IWorkingDaySource"/>.
+        /// </summary>
+        /// <param name="source">The <see cref="IWorkingDaySource"/> to use.</param>
+        /// <returns>The same instance of the <see cref="WorkingDayServiceBuilder"/> using only the <see cref="IWorkingDaySource"/> which was passed in. All all of its current sources will be removed.</returns>
+        public WorkingDayServiceBuilder UseSource(IWorkingDaySource source)
```

Comment:
I don't see the point of `UseSource` vs `AddSource`, `WithSource` would be much simpler to use.
For a single source `WorkingDayService` then you just have a single `WithSource`, then for a multi source `WorkingDayService` you would chain `WithSource` calls.
With `UseSource` and `AddSource` you could create some very weird looking calls.

```
WorkingDayServiceBuilder.New().AddSource(a).AddSource(b).UseSource(c).Build();
```
The `WorkingDayService` would end up only having a source of `c`, but without looking at the comments it would be pretty difficult to understand why.